src/components/metrics.tsx

```diff
@@ -436,6 +439,30 @@ export const QueryTable: React.FC<QueryTableProps> = ({ index, namespace }) => {
     }
   }
 
+  const getCsvColumns = () => {
+    const csvColumns = [];
+    for (let i = 1; i < columns.length; i++) {
+      const column = columns[i];
+      if (typeof column?.title === 'string') {
+        csvColumns.push(column.title);
+      } else if (column.title?.props?.children) {
+        csvColumns.push(column.title.props.children);
+      }
+    }
+    return csvColumns;
+  };
```

Comment:
If we don't find the title in either `column.title` or `column.title?.props?.children`, nothing will be pushed onto `csvColumns`, which will cause the titles to get out of sync with the data.

In other words, I think we need
```
if (typeof column?.title === 'string') {
  csvColumns.push(column.title);
} else if (column.title?.props?.children) {
  csvColumns.push(column.title.props.children);
} else {
  csvColumns.push('');
}
```

Also, I think this is a good case for using a `map`. Especially since using `map` will naturally guarantee that we get one output element for every input element.

So something like below (not tested!)
```
const getCsvColumns = () => {
  return columns.slice(1).map((column) => {
    if (typeof column?.title === 'string') {
      return column.title;
    } else if (column.title?.props?.children) {
      return column.title.props.children;
    } else {
      return '';
    }
  });
};
```

Reply:
Thanks for the suggested function, I tested it and it works! I updated this in the latest commit.